labs/unicorn-stock-broker/src/main/java/com/unicorn/broker/core/UnicornStockBrokerHandler.java

```diff
@@ -0,0 +1,39 @@
+package com.unicorn.broker.core;
+
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.unicorn.broker.model.Transaction;
+import org.springframework.stereotype.Component;
+
+import java.util.function.Function;
+
+@Component
+public class UnicornStockBrokerHandler implements Function<Transaction, APIGatewayProxyResponseEvent> {
+
+    private final TransactionService transactionService;
+
+    public UnicornStockBrokerHandler(TransactionService transactionService) {
+        this.transactionService = transactionService;
+    }
+
+    public APIGatewayProxyResponseEvent apply(final Transaction transaction) {
+        if (!transaction.isValid()) {
+            return createAPIGwResponse(400, "Invalid request body.");
+        }
+
+        var transactionResponse = transactionService.writeTransaction(transaction);
+        return transactionResponse
+                .map(it -> createAPIGwResponse(200, generateJSONResponse(it)))
+                .orElseGet(() ->  createAPIGwResponse(500, "Unexpected error while writing transaction."));
+    }
+
+    private APIGatewayProxyResponseEvent createAPIGwResponse(Integer statusCode, String message){
+        return new APIGatewayProxyResponseEvent()
+                .withStatusCode(statusCode)
+                .withBody(String.format("%s%n", message));
+    }
+
+    private String generateJSONResponse(Transaction transaction) {
+        return String.format("Broker %s successfully created transaction %s", transaction.brokerId.toString().substring(0,8), transaction.transactionId);
```

Comment:
Use a constant for 8 to make clear why a substring is used

Reply:
Added, please check.